File: 107-binary-tree-level-order-traversal-ii/107-binary-tree-level-order-traversal-ii.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
    vector<vector<int>> ans;
public:
    void Levelorder(TreeNode* root, int level){
        if(!root) return;
        if(level==ans.size()) ans.push_back(vector<int>());
        ans[level].push_back(root->val);
        Levelorder(root->left,level+1);
        Levelorder(root->right,level+1);
    }
    
    vector<vector<int>> levelOrderBottom(TreeNode* root) {
        Levelorder(root,0);
        reverse(ans.begin(),ans.end());
        return ans;
    }
};

File: 107-binary-tree-level-order-traversal-ii/107-binary-tree-level-order-traversal-ii.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def Levelorder(self,root,level,ans):
        if(not root): return
        if(level==len(ans)): ans.append([])
        ans[level].append(root.val)
        self.Levelorder(root.left,level+1,ans)
        self.Levelorder(root.right,level+1,ans)
    
    def levelOrderBottom(self, root: Optional[TreeNode]) -> List[List[int]]:
        ans = []
        self.Levelorder(root,0,ans)
        return ans[::-1]